packages/wallet/api/src/lib-wallet.js

```diff
@@ -572,21 +612,62 @@ export function makeWallet({
     /** @type {Map<Payment, Purse>} */
     const paymentToPurse = new Map();
 
+    // Track which gives/wants are attestations.
+    const keywordToAttestation = new Map();
+
     // We now have everything we need to provide Zoe, so do the actual withdrawals.
     // Payments are made for the keywords in proposal.give.
     const keywordPaymentPs = Object.entries(proposal.give || harden({})).map(
-      async ([keyword, amount]) => {
+      async ([keyword, { type, ...amount }]) => {
         const purse = purseKeywordRecord[keyword];
         assert(
           purse !== undefined,
           X`purse was not found for keyword ${q(keyword)}`,
         );
-        const payment = await E(purse).withdraw(amount);
-        paymentToPurse.set(payment, purse);
+        let payment;
```

Comment:
No need for a mutable var. Can just return in each condition. Type system will ensure consistency.

Reply:
Done